metadoc-tests/src/test/scala/metadoc/tests/MetadocCliSuite.scala

```diff
@@ -0,0 +1,354 @@
+package metadoc.tests
+
+import java.nio.file.Files
+import scala.meta.testkit._
+import caseapp.RemainingArgs
+import metadoc.cli.MetadocCli
+import metadoc.cli.MetadocOptions
+import metadoc.{schema => d}
+import org.langmeta.internal.io.FileIO
+import org.langmeta.io.AbsolutePath
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.FunSuite
+
+class MetadocCliSuite
+    extends FunSuite
+    with BeforeAndAfterAll
+    with DiffAssertions {
+  var out: AbsolutePath = _
+  def options = MetadocOptions(
+    Some(out.toString()),
+    cleanTargetFirst = true,
+    nonInteractive = true
+  )
+  def files = BuildInfo.exampleClassDirectory.map(_.getAbsolutePath)
+
+  def runCli(): Unit = MetadocCli.run(options, RemainingArgs(files, Nil))
+
+  override def beforeAll(): Unit = {
+    out = AbsolutePath(Files.createTempDirectory("metadoc"))
+    out.toFile.deleteOnExit()
+    runCli()
+  }
+
+  val expectedFiles =
+    """paiges/core/src/main/scala/org/typelevel/paiges/Chunk.scala.semanticdb
+      |paiges/core/src/main/scala/org/typelevel/paiges/Doc.scala.semanticdb
+      |paiges/core/src/main/scala/org/typelevel/paiges/Document.scala.semanticdb
+      |paiges/core/src/main/scala/org/typelevel/paiges/package.scala.semanticdb
+      |paiges/core/src/test/scala/org/typelevel/paiges/DocumentTests.scala.semanticdb
+      |paiges/core/src/test/scala/org/typelevel/paiges/Generators.scala.semanticdb
+      |paiges/core/src/test/scala/org/typelevel/paiges/JsonTest.scala.semanticdb
+      |paiges/core/src/test/scala/org/typelevel/paiges/PaigesTest.scala.semanticdb
+    """.stripMargin
+  test("target/semanticdb") {
+    val obtained = FileIO
+      .listAllFilesRecursively(out.resolve("semanticdb"))
+      .files
+      .sortBy(_.toString())
+    assertNoDiff(obtained.mkString("\n"), expectedFiles)
+  }
+
+  test("target/symbol") {
+    val obtained = FileIO
+      .listAllFilesRecursively(out.resolve("symbol"))
+      .map { path =>
+        val sym = d.SymbolIndex.parseFrom(path.readAllBytes)
+        assert(sym.definition.isDefined)
+        sym.symbol
+      }
+      .sorted
+      .mkString("\n")
+    val expected =
+      """|
+         |org.typelevel.paiges.Chunk.
+         |org.typelevel.paiges.Chunk.best(Int,Doc,Boolean).
+         |org.typelevel.paiges.Chunk.indentMax.
+         |org.typelevel.paiges.Chunk.indentTable.
+         |org.typelevel.paiges.Chunk.lineToStr(Int).
+         |org.typelevel.paiges.Chunk.makeIndentStr(Int).
+         |org.typelevel.paiges.Doc#
+         |org.typelevel.paiges.Doc#`&:`(String).
+         |org.typelevel.paiges.Doc#`&`(Doc).
+         |org.typelevel.paiges.Doc#`*`(Int).
+         |org.typelevel.paiges.Doc#`+:`(String).
+         |org.typelevel.paiges.Doc#`+`(Doc).
+         |org.typelevel.paiges.Doc#`/:`(String).
+         |org.typelevel.paiges.Doc#`/`(Doc).
```

Comment:
This captures the internal representation for the bug in the UI if I am not mistaken.

The old string read: 

```
|_root_.org.typelevel.paiges.Doc#`/`(Lorg/typelevel/paiges/Doc;)Lorg/typelevel/paiges/Doc;.
```

Reply:
wdym? I did not notice any bugs in the UI

SemanticDB v3 did major changes in the syntax for symbols, in particular methods. The new syntax is more readable and easier to produce outside of the compiler.

Reply:
Aah, I see. We didn't use the symbol syntax before. Before we had `Denotation.signature: String` which was a human readable representation of a type. That string field has been replaced with a `Type` field which is no longer human readable but encodes way more information and should be possible to convert to a more readable format.